Source/Private/PerfGrapherCommandlet.cpp

```diff
@@ -90,23 +108,94 @@ int32 UPerfGrapherCommandlet::Main( const FString & params )
 
     for ( const auto & package_name : package_names )
     {
-        auto * package = LoadPackage( nullptr, *package_name, 0 );
-        if ( package == nullptr )
+        UE_LOG( LogPerfGrapher, Log, TEXT( "Processing package: %s" ), *package_name );
+        if ( !RunPerfGrapher( package_name, metrics_params ) )
         {
-            UE_LOG( LogPerfGrapher, Error, TEXT( "Could not load package %s" ), *package_name );
-            return 2;
+            UE_LOG( LogPerfGrapher, Error, TEXT( "Failed to process map %s" ), *package_name );
+            return 1;
         }
+    }
 
-        auto * world = UWorld::FindWorldInPackage( package );
-        if ( world == nullptr )
-        {
-            UE_LOG( LogPerfGrapher, Error, TEXT( "Could not get a world in the package %s" ), *package_name );
-            return 2;
-        }
+    UE_LOG( LogPerfGrapher, Log, TEXT( "All packages processed successfully" ) );
+    return 0;
+}
 
-        world->WorldType = EWorldType::Editor;
+bool UPerfGrapherCommandlet::RunPerfGrapher( const FString & package_name, const FMetricsParams & metrics_params ) const
+{
+    // :NOTE: Load the world package
+    auto * package = LoadPackage( nullptr, *package_name, LOAD_None );
+    if ( package == nullptr )
+    {
+        UE_LOG( LogPerfGrapher, Error, TEXT( "Cannot load package %s" ), *package_name );
+        return false;
     }
-    return 0;
+    UE_LOG( LogPerfGrapher, Log, TEXT( "Package %s loaded" ), *package_name );
+
+    // :NOTE: Get world from package
+    auto * world = UWorld::FindWorldInPackage( package );
```

Comment:
would be cool to have a RAII class here to handle world init + automatic cleanup when the RAII class is destructed

Reply:
Done, but no code convention about RAII classes so I've done it in a single header file.